Add info about App column, filter, and deployment matching (#2560)
giantswarm/docs: src/content/overview/developer-portal/deployments/index.md

- **Revision last applied** and **Revision last attempted**: for a healthy deployment, these two values should be the same. In a failed deployment, this can help you understand which version failed, and which one is still running instead.
- Deployment time: tells you when the current version has been deployed.
- Errors details in case of a failed deployment: help to save time when troubleshooting a failed deployment. Note that these panels can be expanded and may be collapsed by default.

## Matching deployments and apps in the catalog {#component-matching}

> To provide a link to the app details page, several requirements have to be met:

Reviewer: ```suggestion
To provide a link to the app details page, the system must fulfill several requirements:
```

> - The app must be found in the portal catalog as an entity of kind `Component`.

Author: Implementation changed, let's assume we release this soon

> - The entity found must provide an annotaion `giantswarm.io/deployment-names` with possible deployment names as a comma-separated string. The chart name of the deployment must match one value of the names provided in the annotation.

Reviewer: ```suggestion
- The entity found must provide an annotation `giantswarm.io/deployment-names` with possible deployment names as a comma-separated string. The chart name of the deployment must match one value of the names provided in the annotation.
```
